lib/react.dart

```diff
@@ -775,6 +829,44 @@ abstract class Component2 implements Component {
   List<StateUpdaterCallback> get transactionalSetStateCallbacks => throw new UnimplementedError();
 }
 
+/// Creates a mixin used to ensure consistent typing of the `snapshot`
+/// parameter passed between Component2.getSnapshotBeforeUpdate and
+/// Component2.componentWillUpdate.
+///
+/// __EXAMPLE__:
+///
+/// class _ParentComponent extends react.Component2 implements react
+///    .TypedSnapshot<int> {
+///
+///   // Note that the return value is specified as an int, matching the mixin
+///   // type.
+///   int getShapshotBeforeUpdate(prevProps, prevState) {
+///     if (prevProps["foo"]) {
+///       return true;
+///     }
+///     return null;
+///   }
+///
+///   // Note that the snapshot parameter is consistently typed with the return
+///   // value of getSnapshotBeforeUpdate.
+///   void componentDidUpdate(prevProps, prevState, [int snapshot]) {
+///     if (snapshot == true) {
+///       //Do something
```

Comment:
#nit add spaces after comments
```suggestion
///       // Do something
```